src/main/java/javaslang/collection/Vector.java

```diff
@@ -100,6 +100,39 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
     }
 
     /**
+     * Returns a Vector containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param <T> Component type of the Vector
+     * @param n The number of elements in the Vector
+     * @param f The Function computing element values
+     * @return A Vector consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code n} or {@code f} are null
+     */
+    public static <T> Vector<T> fill(Integer n, Function<Integer, ? extends T> f) {
+        Objects.requireNonNull(n, "n is null");
+        Objects.requireNonNull(f, "f is null");
+        HashArrayMappedTrie<Integer, T> trie = HashArrayMappedTrie.empty();
+        for (int i = 0; i <n; i++) {
+            trie = trie.put(trie.size(), f.apply(i));
+        }
+        return new Vector<>(trie);
```

Comment:
Ok, got it. So is ofAll(Iterable) broken with respect to this semantic? The following test is red
```@Test
public void t() {
    Object[] c = new Object[0];
    Vector<Object> v = Vector.ofAll(Array.of(c));
    assertThat(v).isSameAs(Vector.empty());
}
```